src/PostgREST/Workers.hs

```diff
@@ -208,6 +209,7 @@ listener appState = do
     handleFinally dbChannel _ = do
       -- if the thread dies, we try to recover
       AppState.logWithZTime appState $ "Retrying listening for notifications on the " <> dbChannel <> " channel.."
+      AppState.putIsListenerOn appState False
```

Comment:
Codecov warning should be gone when https://github.com/PostgREST/postgrest/issues/1766#issuecomment-999241249 is done.